src/database/queries/entityType.js

```diff
@@ -159,4 +159,48 @@ module.exports = class UserEntityData {
 			return error
 		}
 	}
+
+	static async deleteEntityTypesAndEntities(filter) {
+		try {
+			// Step 1: Find all entityTypes where the filter conditions are met (e.g., status is ACTIVE and certain values in 'value' column)
+			const entityTypes = await EntityType.findAll({
+				where: filter,
+				raw: true,
+			})
+
+			const entityTypeIds = entityTypes.map((entityType) => entityType.id)
+
+			if (entityTypeIds.length > 0) {
+				// Step 2: Fetch all matching entities using the entityType IDs
+				const entities = await Entity.findAll({
+					where: { entity_type_id: entityTypeIds, status: 'ACTIVE' },
+					raw: true,
+				})
+
+				// Step 3: Prepare result with entityTypes and their associated entities
+				const result = entityTypes.map((entityType) => {
+					const matchingEntities = entities.filter((entity) => entity.entity_type_id === entityType.id)
+					return {
+						...entityType,
+						entities: matchingEntities,
+					}
+				})
+
+				// Step 4: Delete the entities and entityTypes
+				await Entity.destroy({
+					where: { entity_type_id: entityTypeIds },
+				})
+
+				await EntityType.destroy({
+					where: { id: entityTypeIds },
+				})
+				return result
```

Comment:
Why are we manually deleting the entity types? Can you refer to the existing flow, where we are using hooks to delete the entities?
Can you check if it's possible here?

Reply:
Instead of 
```
				await Entity.destroy({
					where: { entity_type_id: entityTypeIds },
				})
```
can we make use of hooks here?